src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/QilGenerator.cs

```diff
@@ -196,13 +196,14 @@ private QilExpression Compile(Compiler compiler)
             }
 
             // Create list of all early bound objects
-            Dictionary<string, Type?> scriptClasses = compiler.Scripts.ScriptClasses;
+            Scripts.LinkerSafeDictionary scriptClasses = compiler.Scripts.ScriptClasses;
```

Comment:
Nit: Try to not use the term `link` outside of the linker repo - it's confusing since it typically means different thing (platform linker). We try to stick to the term `trim`/`trimmer` everywhere.